### ISSUE TITLE
Fails to typecheck with flow 0.54.1

### ISSUE THREAD

**Issue by jfirebaugh** [rpl/flow-coverage-report#119]
When using flow in a project where flow-coverage-report is installed to node_modules, I get the following errors:

```
Error: node_modules/flow-coverage-report/src/lib/report-html.js:70
                v------------------------
 70:     return Promise.reject(new Error(
 71:       'filename is not relative to the projectDir: ' +
 72:         [opt.projectDir, opt.filename].join(' - ')
 73:     ));
         -^ type parameter `T` of call of method `reject`. Missing annotation

Error: node_modules/flow-coverage-report/src/lib/report-html.js:181
                v-----------------------------------------
181:     return waitForReportContent.then(function (res) {
182:       var reportFilePath = res.reportFilePath;
183:       var reportFileContent = res.reportFileContent;
...:
188:     });
         -^ type parameter `U` of call of method `then`. Missing annotation

Error: node_modules/flow-coverage-report/src/lib/report-html.js:191
191:   return Promise.reject(new Error('Unknown report type: ' + opt.type));
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ type parameter `T` of call of method `reject`. Missing annotation
```

I've excluded flow-coverage-report via my .flowconfig for now, but it would be nice to have this fixed upstream.

**Comment by julienw** [comment]
I have a PR for this for 2 weeks: https://github.com/rpl/flow-coverage-report/pull/116

although I'm not sure why Flow is not happy with this, as I don't reproduce the issue on a simpler example :/

**Comment by julienw** [comment]
I filed a bug at Flow: https://github.com/facebook/flow/issues/4923

My guess is Flow starts infering by looking at how a method is called, and then checks whether it works with the existing code. But in a library we can't know how it's called so it can't infer properly.

That's why the fix is to help him by typing the return values.

**Comment by rpl** [comment]
#116 has been merged in #125 (and released on npm in the new 0.4.0 version).
